### PR TITLE
PHPDoc and tests cleanup

### PR DESCRIPTION
**Pull request by thePanz**

### CODE REVIEW

File: lib/Elastica/Cluster.php [hunk 1]
```diff
@@ -138,8 +127,6 @@ public function getClient()
      * @see https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-nodes-info.html
      *
      * @param array $args Additional arguments
-     *
-     * @throws \Elastica\Exception\NotImplementedException
```
Comment (ruflin) [inline lib/Elastica/Cluster.php]: Isn't that still needed? Or is this replaced by some "magic"?

File: lib/Elastica/Node.php [hunk 2]
```diff
@@ -34,55 +34,43 @@ class Node
...
-     *
-     * @return $this Refreshed object
-     */
-    public function setId($id)
+    public function setId(string $id): self
     {
         $this->_id = $id;
+        $this->refresh();
 
-        return $this->refresh();
```
Comment (ruflin) [inline lib/Elastica/Node.php]: Nit: Does it have to be 2 lines?

File: test/Elastica/Base.php [hunk 3]
```diff
@@ -22,41 +22,6 @@ protected static function showDeprecated()
         \error_reporting(\error_reporting() | E_USER_DEPRECATED);
     }
 
-    protected function assertFileDeprecated($file, $deprecationMessage)
```
Comment (ruflin) [inline test/Elastica/Base.php]: Quick reminder for me: We don't need this anymore because ...?

File: test/Elastica/Base.php [hunk 4]
```diff
@@ -104,20 +69,21 @@ protected function _getProxyUrl403(): string
...
-        if (null === $name) {
-            $name = \preg_replace('/[^a-z]/i', '', \strtolower(\get_called_class()).\uniqid());
-        }
+        $name = $name ?: static::buildUniqueId();
 
         $client = $this->_getClient();
-        $index = $client->getIndex('elastica_'.$name);
+        $index = $client->getIndex($name);
 
-        if ('elasticsearch' === \getenv('ES_HOST')) {
```
Comment (ruflin) [inline test/Elastica/Base.php]: This is not needed anymore?

File: test/Elastica/ClusterTest.php [hunk 5]
```diff
@@ -3,74 +3,63 @@
 namespace Elastica\Test;
 
 use Elastica\Cluster;
-use Elastica\Cluster\Health;
 use Elastica\Node;
 use Elastica\Test\Base as BaseTest;
 
+/**
+ * @group functional
```
Comment (ruflin) [inline test/Elastica/ClusterTest.php]: Nice, didn't know you can set it on the class level

File: test/Elastica/Processor/AttachmentTest.php [hunk 6]
```diff
@@ -66,7 +66,7 @@ public function testAttachmentAddPdf()
         $bulk->setIndex($index);
 
         $doc1 = new Document(null);
-        $doc1->addFile('data', BASE_PATH.'/data/test.pdf');
+        $doc1->addFile('data', __DIR__.'/../../data/test.pdf');
```
Comment (ruflin) [inline test/Elastica/Processor/AttachmentTest.php]: Much nicer.

File: test/Elastica/ResponseFunctionalTest.php [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+use Elastica\Mapping;
+use Elastica\Query;
+use Elastica\Query\MatchAll;
+use Elastica\Request;
+use Elastica\Test\Base as BaseTest;
+
+/**
+ * @group functional
+ */
+class ResponseFunctionalTest extends BaseTest
```
Comment (ruflin) [inline test/Elastica/ResponseFunctionalTest.php]: Always happy to have more tests. Could you share some background if this is "in addition" or replacing something else?

File: test/Elastica/ResponseTest.php [hunk 8]
```diff
@@ -2,82 +2,15 @@
...
 
+/**
+ * @group unit
+ */
 class ResponseTest extends BaseTest
 {
-    /**
-     * @group functional
-     */
-    public function testResponse()
```
Comment (ruflin) [inline test/Elastica/ResponseTest.php]: Ah, the other one is replacing this part?

File: test/Elastica/Processor/AttachmentTest.php (hunk 6 above)
Comment (thePanz) [reply test/Elastica/Processor/AttachmentTest.php]: :)

File: test/Elastica/ResponseTest.php (hunk 8 above)
Comment (thePanz) [reply test/Elastica/ResponseTest.php]: yes, exactly. :)

File: test/Elastica/ResponseFunctionalTest.php (hunk 7 above)
Comment (thePanz) [reply test/Elastica/ResponseFunctionalTest.php]: Those are tests extracted from the ResponseTest

File: test/Elastica/Base.php (hunk 3 above)
Comment (thePanz) [reply test/Elastica/Base.php]: This method was not used

File: lib/Elastica/Cluster.php (hunk 1 above)
Comment (thePanz) [reply lib/Elastica/Cluster.php]: This method is just a placeholder, as it is only throwing that exception.
I was in doubt to remove the method completely ... wdyt?

**Comment by p365labs** [comment]
@thePanz is the PHPDOC a cleanup due to the ```NoSuperfluousPhpdocTags``` introduction inside phpcsfixer (v.2.16.0)?

File: lib/Elastica/Cluster.php (hunk 1 above)
Comment (ruflin) [reply lib/Elastica/Cluster.php]: I think my "idea" here was that if users see an method and get an exception with "not implemented" they are likely to contribute it. But seems like it never happened. I would say either we keep the docs entry or remove the method completly. I'm good with removing it completely. I guess there are 2-3 other similar methods.

**Comment by thePanz** [comment]
@p365labs yes, that's what I a aiming to. I am manually cleaning up the code, and discovering some interesting cases (methods where the PHP docs is `return $this`, but the method does not return anything).

I will open another PR with the `NoSuperfluousPhpdocTags` flag enabled :)

File: lib/Elastica/Cluster.php (hunk 1 above)
Comment (thePanz) [reply lib/Elastica/Cluster.php]: Removed the "not-implemented" code!

File: lib/Elastica/Node.php (hunk 2 above)
Comment (ruflin) [reply lib/Elastica/Node.php]: I see, need to learn the rules :-D

**Comment by thePanz** [comment]
@ruflin CI is now green, should we merge this? :)

**Comment by thePanz** [comment]
@ruflin do you mean adding a changelog about never-implemented functions?

I believe it is bringing more confusion to the users reading it, and no more clarity

**Pull request merged**